code/game/atoms_movable.dm

```diff
@@ -268,11 +268,12 @@
 	//If no destination, move the atom into nullspace (don't do this unless you know what you're doing)
 	else
 		. = TRUE
-		var/atom/oldloc = loc
-		var/area/old_area = get_area(oldloc)
-		oldloc.Exited(src, null)
-		if(old_area)
-			old_area.Exited(src, null)
+		if (loc)
+			var/atom/oldloc = loc
+			var/area/old_area = get_area(oldloc)
+			oldloc.Exited(src, null)
+			if(old_area)
+				old_area.Exited(src, null)
 		loc = null

```

Comment:
`moveToNullSpace` here aswell?

Reply:
No, this is literally what moveToNullspace calls.